### DISCUSSION TITLE
Bundling flow without // @flow pragma

### DISCUSSION THREAD

**Discussion by Brianzchen** [parcel-bundler/parcel#5524] [Q&A] [answered]
I've swapped from using `// @flow` to adding a licence header to the top of my files with a flow pragma embedded. Because of this, parcel can't understand this is a flow file and crashes when it finds a line of code such as `(): Reat.Node`

My header looks something like this
```
/**
 * Some license stuff here
 *
 * @flow
 */
```

For reference this is a library that runs with `babel.config.js` and I'm only using parcel to bundle a documentation website. Swapping out my babel config is out of the question.

How can I force parcel to run whatever flow parser it has either manually or any way that doesn't require a `.babelrc`?

FYI: I'm running `"parcel-bundler": "^1.12.4"`

**Comment by mischnic** [comment, marked as answer]
This is the logic for Parcel 1:  https://github.com/parcel-bundler/parcel/blob/96cdc44690688a6dd7f0943d14022f3c4ab57503/packages/core/parcel-bundler/src/transforms/babel/flow.js#L5

So the only way in your case is
```
/** @flow
 *
 * Some license stuff here
 *
 */
```

Parcel 2 instead adds the babel plugin if `flow-bin` is listed as a dependency in package.json. (and would also read your babel config anyway)

> whatever flow parser it

~parser~ babel plugin: https://github.com/parcel-bundler/parcel/blob/96cdc44690688a6dd7f0943d14022f3c4ab57503/packages/core/parcel-bundler/src/transforms/babel/config.js#L81

**Reply by Brianzchen** [reply]
Thanks for the quick reply! Works perfectly
